client_test.go

```diff
@@ -110,20 +106,26 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
-func waitForDaemonStart(ctx context.Context, client *Client) error {
+func waitForDaemonStart(ctx context.Context, address string) (*Client, error) {
 	var (
+		client  *Client
 		serving bool
 		err     error
 	)
 
 	for i := 0; i < 20; i++ {
-		serving, err = client.IsServing(ctx)
-		if serving {
-			return nil
+		if client == nil {
```

Comment:
if the dial fails (which it appears could have happened prior, but now "forced" more likely with the block/fail if not temporary options), yes, client will be `nil`. So now `waitForDaemonStart` has the retry loop to handle this case--at least that's my understanding